samtranslator/schema/schema.py

```diff
@@ -52,42 +51,86 @@ class AwsServerlessConnector(BaseModel):
+    Key: Union[str, SamIntrinsic]
+    Version: Optional[Union[str, SamIntrinsic]]
+
+
+class FunctionDeploymentPreferenceHooks(BaseModel):
+    PostTraffic: Optional[Union[str, SamIntrinsic]]
+    PreTraffic: Optional[Union[str, SamIntrinsic]]
+
+
+class FunctionDeploymentPreference(BaseModel):
+    Alarms: Optional[Union[List[SamIntrinsic], SamIntrinsic]]
+    Enabled: Optional[Union[bool, SamIntrinsic]]
+    Hooks: Optional[FunctionDeploymentPreferenceHooks]
+    PassthroughCondition: Optional[Union[bool, SamIntrinsic]]
+    Role: Optional[Union[str, SamIntrinsic]]
+    TriggerConfigurations: Optional[PassThrough]
+    Type: Optional[
+        Union[str, SamIntrinsic]
+    ]  # TODO: Should investigate whether this is a required field. This is a required field on documentation. However, we don't seem to use this field.
+
+
+class FunctionDeadLetterQueue(BaseModel):
+    TargetArn: str
+    Type: Literal["SNS", "SQS"]
+
+
+class FunctionEventInvokeConfiguration(BaseModel):
+    DestinationConfig: Optional[PassThrough]
+    MaximumEventAgeInSeconds: Optional[int]
+    MaximumRetryAttempts: Optional[int]
+
+
+class FunctionEvent(BaseModel):
+    Properties: Any
+    Type: str
+
+
+class FunctionUrlConfig(BaseModel):
+    AuthType: Union[str, SamIntrinsic]
+    Cors: Optional[PassThrough]
+
+
 class FunctionProperties(BaseModel):
-    Architectures: Unknown
-    AssumeRolePolicyDocument: Unknown
-    AutoPublishAlias: Unknown
-    AutoPublishCodeSha256: Unknown
-    CodeSigningConfigArn: Unknown
-    CodeUri: Unknown
-    DeadLetterQueue: Unknown
-    DeploymentPreference: Unknown
-    Description: Unknown
-    Environment: Unknown
-    EphemeralStorage: Unknown
-    EventInvokeConfig: Unknown
-    Events: Unknown
-    FileSystemConfigs: Unknown
-    FunctionName: Unknown
-    FunctionUrlConfig: Unknown
-    Handler: Unknown
-    ImageConfig: Unknown
-    ImageUri: Unknown
-    InlineCode: Unknown
-    KmsKeyArn: Unknown
-    Layers: Unknown
-    MemorySize: Unknown
-    PackageType: Unknown
-    PermissionsBoundary: Unknown
-    Policies: Unknown
-    ProvisionedConcurrencyConfig: Unknown
-    ReservedConcurrentExecutions: Unknown
-    Role: Unknown
-    Runtime: Unknown
-    Tags: Unknown
-    Timeout: Unknown
-    Tracing: Unknown
-    VersionDescription: Unknown
-    VpcConfig: Unknown
+    Architectures: Optional[PassThrough]
+    AssumeRolePolicyDocument: Optional[Dict[str, Any]]
+    AutoPublishAlias: Optional[Union[str, SamIntrinsic]]
+    AutoPublishCodeSha256: Optional[Union[str, SamIntrinsic]]
+    CodeSigningConfigArn: Optional[Union[str, SamIntrinsic]]
+    CodeUri: Optional[Union[str, FunctionCodeUri]]
+    DeadLetterQueue: Optional[Union[SamIntrinsic, FunctionDeadLetterQueue]]
+    DeploymentPreference: Optional[FunctionDeploymentPreference]
+    Description: Optional[PassThrough]
+    Environment: Optional[PassThrough]
+    EphemeralStorage: Optional[PassThrough]
+    EventInvokeConfig: Optional[FunctionEventInvokeConfiguration]
```

Comment:
nit: we'll have a lot of names, so keeping it consistent throughout will make it easier to reason (i.e. use `Config` if that's what the property uses)